src/tools/rbd/action/Import.cc

```diff
@@ -387,6 +387,9 @@ static int do_import_header(int fd, int import_format, uint64_t &size, librbd::I
 	}
       } else {
 	std::cerr << "rbd: invalid tag in image properties zone: " << tag << "Skip it." << std::endl;
+	if (fd == 0) {
+	  return -EINVAL;
+	}
 	::lseek(fd, length, SEEK_CUR);
```

Comment:
@yangdongsheng You can use `read` instead of `lseek` for stdin. Also it is better to use a constant `STDIN_FILENO` instead of 0. And you need to check `lseek` return value (probably falling back to `read` if `errno` is set to `ESPIPE` and failing for any other error).